sysdeps/aero/generic/filesystem.cpp

```diff
@@ -0,0 +1,276 @@
+#include <mlibc/debug.hpp>
+
+namespace mlibc {
+int sys_write(int fd, const void *buffer, size_t count, ssize_t *written) {
+    auto result = syscall(SYS_WRITE, fd, buffer, count);
+
+    if (result < 0) {
+        return -result;
+    }
+
+    *written = result;
+    return 0;
+}
+
+int sys_read(int fd, void *buf, size_t count, ssize_t *bytes_read) {
+    auto result = syscall(SYS_READ, fd, buf, count);
+
+    if (result < 0) {
+        *bytes_read = 0;
+        return -result;
+    }
+
+    *bytes_read = result;
+    return 0;
+}
+
+int sys_pwrite(int fd, const void *buffer, size_t count, off_t off, ssize_t *written) UNIMPLEMENTED("sys_pwrite")
+int sys_pread(int fd, void *buf, size_t count, off_t off, ssize_t *bytes_read) UNIMPLEMENTED("sys_pread")
+
+int sys_seek(int fd, off_t offset, int whence, off_t *new_offset) {
+    auto result = syscall(SYS_SEEK, fd, offset, whence);
+
+    if (result < 0) {
+        return -result;
+	}
+
+    *new_offset = result;
+    return 0;
+}
+
+int sys_open(const char *filename, int flags, int *fd) {
+    auto result = syscall(SYS_OPEN, 0, filename, strlen(filename), flags);
+
+    if (result < 0) {
+        return -result;
+    }
+
+    *fd = result;
+    return 0;
+}
+
+int sys_close(int fd) {
+    auto result = syscall(SYS_CLOSE, fd);
+
+    if (result < 0) {
+        return -result;
+    }
+
+    return 0;
+}
+
+int sys_access(const char *filename, int mode) {
+    auto result =
+        syscall(SYS_ACCESS, AT_FDCWD, filename, strlen(filename), mode, 0);
+
+    if (result < 0) {
+        return -result;
+    }
+
+    return 0;
+}
+
+int sys_stat(fsfd_target fsfdt, int fd, const char *path, int flags,
+             struct stat *statbuf) {
+    mlibc::infoLogger() << "sys_stat() is unimplemented" << frg::endlog;
+
+    memset(statbuf, 0, sizeof(struct stat));
+    return 0;
+}
+
+int sys_ioctl(int fd, unsigned long request, void *arg, int *result) {
+    auto sys_res = syscall(SYS_IOCTL, fd, request, arg);
+
+    if (sys_res < 0) {
+        return -sys_res;
+    }
+
+    *result = sys_res;
+    return 0;
+}
+
+int sys_isatty(int fd) {
+    // NOTE: The easiest way to check if a file descriptor is a TTY is to
+    // do an ioctl of TIOCGWINSZ on it and see if it succeeds :^)
+    struct winsize ws;
+    int result;
+
+    if (!sys_ioctl(fd, TIOCGWINSZ, &ws, &result)) {
+        return 0;
+    }
+
+    return ENOTTY;
+}
+
+int sys_tcgetattr(int fd, struct termios *attr) {
+    int result;
+
+    if (int e = sys_ioctl(fd, TCGETS, (void *)attr, &result); e)
+        return e;
+
+    return 0;
+}
+
+int sys_tcsetattr(int fd, int optional_action, const struct termios *attr) {
+    if (optional_action)
+        mlibc::infoLogger()
+            << "mlibc: warning: sys_tcsetattr ignores optional_action"
+            << frg::endlog;
+
+    int result;
+
+    // 0x5404 = TCSETSF (using this workaround since the headers don't define
+    // it)
+    if (int e = sys_ioctl(fd, 0x5404, (void *)attr, &result); e)
```

Comment:
After #368, you can drop the hardcoded hack and use the actual define.

Reply:
#368 is merged.

Reply:
done :^)